pkg/api/node_api.go

```diff
@@ -916,6 +918,141 @@ func (a *NodeApi) snapshotStateHash(w http.ResponseWriter, r *http.Request) erro
 	return nil
 }
 
+type GeneratorInfo struct {
+	Address       string `json:"address"`
+	Balance       uint64 `json:"balance"`
+	TransactionID string `json:"transactionID"`
+}
+
+func (a *NodeApi) GeneratorsAt(w http.ResponseWriter, r *http.Request) error {
+	heightStr := chi.URLParam(r, "height")
+	height, err := strconv.ParseUint(heightStr, 10, 64)
+	if err != nil {
+		return errors.Wrap(err, "invalid height")
+	}
+
+	activationHeight, err := a.state.ActivationHeight(int16(settings.DeterministicFinality))
+	if err != nil {
+		return fmt.Errorf("failed to get DeterministicFinality activation height: %w", err)
+	}
```

Comment:
What if activation height is unknown at the moment? I think the endpoint returns `500` error in such case.